parameter/criterion/include/criterion/Criterion.h

```diff
@@ -0,0 +1,180 @@
+
+    virtual bool isInclusive() const override;
+
+    bool getLiteralValue(int numericalValue, std::string& literalValue) const override final;
+
+    virtual bool getNumericalValue(const std::string& literalValue,
+                                   int& numericalValue) const override final;
+
+    virtual std::string getFormattedState() const override;
+    // @}
+
+    bool hasBeenModified() const;
+    void resetModifiedStatus();
+
+    /** Request criterion state match with a desired method
+     *
+     * @param[in] method, the desired match method
+     * @param[in] state, the state to match
+     * @return true if the current state match the state given in parameter with the desired method
+     */
+    bool match(const std::string& method, const State& state) const;
+
+    /** Check if a match method is available for this criterion
+     *
+     * @param[in] method, the desired match method
+     * @return true if the method is available, false otherwise
+     */
+    bool isMatchMethodAvailable(const std::string& method) const;
+
+    std::string getFormattedDescription(bool bWithTypeInfo, bool bHumanReadable) const;
+
+    /** List different values a criterion can have
+     *
+     * @return formatted string containing criterion possible values
+     */
+    std::string listPossibleValues() const;
+
+    /** Export to XML
+     *
+     * @param[in] xmlElement The XML element to export to
+     * @param[in] serializingContext The serializing context
+     */
+    virtual void toXml(CXmlElement& xmlElement, CXmlSerializingContext& serializingContext) const;
+
+protected:
+    /** Criterion Match callback type
+     *
+     * Those method should take an integer in parameter which represents the state to match
+     * and returns a boolean which indicates if the current state match the state given in
+     * parameter.
+     */
+    typedef std::function<bool (const State&)> MatchMethod;
+
+    /** Match method container, MatchMethod are indexed by their name */
+    typedef std::map<std::string, MatchMethod> MatchMethods;
+
+    /** Initializer constructor
+     * This Constructor initialize class members and should be called by derived class
+     * in order to add functionalities
+     *
+     * @param[in] name the criterion name
+     * @param[in] logger the main application logger
+     * @param[in] derivedValuePairs initial value pairs of derived classes
+     * @param[in] derivedMatchMethods match methods of derived classes
+     */
+    Criterion(const std::string& name,
+              core::log::Logger& logger,
+              const Values& derivedValuePairs,
+              const State&& defaultState,
+              const MatchMethods& derivedMatchMethods);
+
+    /** Contains pair association between literal and numerical value */
+    const Values mValues;
+
+    /** Available criterion match methods */
+    const MatchMethods mMatchMethods;
+
+    /** Current state */
+    State mState;
```

Comment:
state is protected because the derived class handle it differently, so maybe we need a third class which will be abstract ?